src/rgw/rgw_kafka.cc

```diff
@@ -79,40 +118,29 @@ struct connection_t {
   // cleanup of all internal connection resource
   // the object can still remain, and internal connection
   // resources created again on successful reconnection
-  void destroy(int s) {
-    status = s;
-    // destroy temporary conf (if connection was never established)
-    if (temp_conf) {
-        rd_kafka_conf_destroy(temp_conf);
-        return;
-    }
-    if (!is_ok()) {
+  void destroy() {
+    if (!producer) {
       // no producer, nothing to destroy
       return;
     }
-    // wait for all remaining acks/nacks
```

Comment:
you still need to call `rd_kafka_flush` to flush out the remaining callbacks

Reply:
I'm clearing the remaining callbacks (with an error message) in line 133. so there is no issue with clearing the callbacks.
if the broker is ok and shutdown process is called due to a reload (since  https://github.com/ceph/ceph/pull/56979 is already merged) `rd_kafka_flush()` will invoke the callbacks with success.
OTOH, if the broker is not ok, it may delay the reload process in another 5 seconds, so I'm not sure it is a good idea.

Reply:
> if the broker is ok and shutdown process is called d

nope, i was talking in context of shutdown, thats the only other route from where the `destroy` will now be invoked other than `IDLE`, so in shutdown case we just wait additional 5 seconds for callbacks to process, should be ok ?

Reply:
i'll try to change and test

Reply:
added back flushing.

Reply:
but reduced to 500ms (compromise between not blocking reloads to letting the kafka broker reply correctly)